test/unit/puppet_fact_parser_test.rb

```diff
@@ -233,8 +234,35 @@ def setup
     assert_equal '172.30.43.0', parser.interfaces['local_area_connection_2'][:network]
   end
 
+  test "should create connection between new os and new arch on facts parsing" do
+    assert os_to_arch_connectivity_check
+  end
+
+  test "should create connection between existing os and new arch on facts parsing" do
+    FactoryGirl.create(:operatingsystem,:name => 'AnyOS', :major => '6')
+    assert os_to_arch_connectivity_check
```

Comment:
Good point!  Perhaps just call both methods in the test.. or this points to an issue that we should actually be doing this in Host::Base or Host::Managed, in the populate_fields_from_facts method.  An advantage of doing it there would be that this would equally apply to other parsers, such as Chef or Salt.  @ares, what do you think?